contrib/cmake/build3rdParty/build3rdParty.cpp

```diff
@@ -0,0 +1,273 @@
+                    fs::remove_all(vcpkgDir / "ports" / portname);
+                }
+                if (portversion.size() == 40)
+                {
+                    fs::current_path(cloneDir);
+                    execute("git checkout --quiet " + portversion);
+                    cout << "Copying port for " << portname << " from vcpkg commit " << portversion << endl;
+                    fs::copy(cloneDir / "ports" / portname, vcpkgDir / "ports" / portname, fs::copy_options::recursive);
+                    fs::current_path(vcpkgDir);
+                }
+                else
+                {
+                    cout << "Copying port for " << portname << " from SDK customized port " << portversion << endl;
+                    fs::copy(sdkRootPath / "contrib" / "cmake" / "vcpkg_extra_ports" / portname / portversion, vcpkgDir / "ports" / portname, fs::copy_options::recursive);
+                }
+            }
+
+            if (removeUnusedPorts)
+            {
+                for (auto dir = fs::directory_iterator(vcpkgDir / "ports"); dir != fs::directory_iterator(); ++dir)
+                {
+                    if (ports.find(dir->path().filename().u8string()) == ports.end())
+                    {
+                        fs::remove_all(dir->path());
+                    }
+                }
+            }
+
+        }
+        else if (build)
+        {
+
+            if (!fs::is_directory("vcpkg"))
+            {
+                cout << "This command should be run from just outside 'vcpkg' folder - maybe it is not set up?" << endl;
+                return 1;
+            }
+            else
+            {
+                fs::current_path("vcpkg");
+            }
+
+
+            for (auto portPair : ports)
+            {
+                #ifdef WIN32
+                    execute("vcpkg install --triplet " + triplet + " " + portPair.first);
+                #else
+                    execute("./vcpkg install --triplet " + triplet + " " + portPair.first);
+                #endif
+            }
+        }
+
+
+    }
+
+    return 0;
+}
+catch (exception& e)
+{
+    cout << "exception: " << e.what() << endl;
+    return 1;
+}
+
+
+void execute(string command)
+{
+    cout << "Executing: " << command << endl;
+    int result = system(command.c_str());
+    if (result != 0)
+    {
+        cout << "Command failed with resuld code " << result << " ( command was " << command << ")" <<endl;
+        exit(1);
+    }
+}
+
+
+bool showSyntax()
+{
+    cout << "build3rdParty --setup [--removeunusedports] --ports <ports override file> --triplet <triplet> --sdkroot <path>" << endl;
+    cout << "build3rdParty --build --ports <ports override file> --triplet <triplet>" << endl;
+    return false;
+}
+
+bool readCommandLine(int argc, char* argv[])
+{
+    if (argc <= 1) return showSyntax();
+
+    std::vector<char*> myargv1(argv + 1, argv + argc);
+    std::vector<char*> myargv2;
```

Comment:
Based on what happens below, looks like `unrecognizedArgs` might be a better name for `myargv2`

Reply:
no objection :-)